### PR TITLE
[PT-1.13] Upgrade SMModelParallel in Training SM DLC, Fix PT2.0 breaking changes

### PR DESCRIPTION
**Pull request by ShiboXing**
*GitHub Issue #, if available:*

**Note**: 
- If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

- All PR's are checked weekly for staleness. This PR will be closed if not updated in 30 days.

### Description
1. upgrade Rubik binary in PT1.13
2. fix breaking changes from PT2.0
### Tests run

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [ ] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

### Formatting
- [ ] I have run `black -l 100` on my code (formatting tool: https://black.readthedocs.io/en/stable/getting_started.html)

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ShiboXing** [comment]
sm tests:
efa: da370e1
standard: b316cbe
rc: 2351d4f

File: pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu [hunk 1]
```diff
@@ -486,7 +486,8 @@ RUN pip install --no-cache-dir -U \
     "sagemaker>=2,<3" \
     sagemaker-experiments==0.* \
     sagemaker-pytorch-training \
-    flash-attn==0.2.6.post1
+    flash-attn==0.2.8 \
+    triton==2.0.0.dev20221202
```
Comment (shantanutrip) [inline pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu:490]: Why do we need triton?

File: src/requirements.txt [hunk 2]
```diff
@@ -5,6 +5,6 @@ pyfiglet==0.8.post1
 reprint==0.5.2
 ruamel.yaml==0.16.7
 boto3==1.11.11
-black==23.3.0
+black
```
Comment (shantanutrip) [inline src/requirements.txt:8]: Why do we need to unpin this?

File: test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-adapt-BERT_pytorch-training-with-gloo-mpi-backend.patch [hunk 3]
```diff
@@ -1,14 +1,14 @@
-From 041d5935dfdb0b3b0ee8ce38afbe528ffa29e58c Mon Sep 17 00:00:00 2001
-From: Your Name <you@example.com>
```
Comment (shantanutrip) [inline test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-adapt-BERT_pytorch-training-with-gloo-mpi-backend.patch:2]: Is this change required?

File: pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu (hunk 1 above)
Comment (ShiboXing) [reply pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu:490]: Although openai's triton is directly depended by PyTorch, Rubik team currently requires a specific build of triton, hence the explicit installation.

File: src/requirements.txt (hunk 2 above)
Comment (ShiboXing) [reply src/requirements.txt:8]: I could revert this. I thought that since we have added the style check to the PR, local black changes might differ from the CICD test if pinned for too long.

File: test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-adapt-BERT_pytorch-training-with-gloo-mpi-backend.patch (hunk 3 above)
Comment (ShiboXing) [reply test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-adapt-BERT_pytorch-training-with-gloo-mpi-backend.patch:2]: After updating the patch to be applied to torch repo, I overwrote the entire fie, hence the change.

File: test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-adapt-BERT_pytorch-training-with-gloo-mpi-backend.patch [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
-Date: Thu, 6 Apr 2023 23:00:12 +0000
-Subject: [PATCH] test: adapt BERT_pytorch training with gloo and mpi backend
+From f97dbc04afbaccd14bf27f23c3a3b066e44e456d Mon Sep 17 00:00:00 2001
+From: Shibo Xing <shibox@amazon.com>
```
Comment (jeet4320) [inline test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-adapt-BERT_pytorch-training-with-gloo-mpi-backend.patch:2]: do we need name here?

File: test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-add-flag-for-inductor-in-ddp.patch [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
++        'resnet50': 64,
+     }
+     model_batch_size = {MODEL_PATH_TEMPLATE.format(k): v for k, v in model_batch_size.items()}
+     # put eager first to ensure it can be used for reference values.
+     # try --torchdynamo eager or --torchdynamo aot_eager for debugging
+     model_args_configs = [
+         [],  # no args = pure eager baseline
+-        ["--torchdynamo", "inductor"],
+     ]
++    if os.environ["USE_INDUCTOR"] == '1':
```
Comment (jeet4320) [inline test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-add-flag-for-inductor-in-ddp.patch:28]: how much time inductor tests added to total test duration?

File: test/dlc_tests/container_tests/bin/pytorch_tests/testPTHVD [hunk 6]
```diff
@@ -6,5 +6,6 @@ LOG_DIR=${HOME_DIR}/logs
 HOROVOD_VERSION=v0.16.4
 
 git clone -b ${HOROVOD_VERSION} https://github.com/horovod/horovod.git ${HOME_DIR}/artifacts/horovod
+pip uninstall -y smdistributed_modelparallel
```
Comment (jeet4320) [inline test/dlc_tests/container_tests/bin/pytorch_tests/testPTHVD:9]: why to uninstall it

**Review by jeet4320 — CHANGES_REQUESTED** [review]

File: test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-adapt-BERT_pytorch-training-with-gloo-mpi-backend.patch (hunk 4 above)
Comment (ShiboXing) [reply test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-adapt-BERT_pytorch-training-with-gloo-mpi-backend.patch:2]: sorry, I used my email info for git format-patch by mistake. I have another PR where I can immediately redo the format-patch with a default git config.

File: test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-add-flag-for-inductor-in-ddp.patch (hunk 5 above)
Comment (ShiboXing) [reply test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-add-flag-for-inductor-in-ddp.patch:28]: This test takes only ~10 minutes. The bulk of the added duration is in the benchmark coedbuild. After removals of some oudated tests, EC2 and SM codebuils currently take about the same time as before.

File: test/dlc_tests/container_tests/bin/pytorch_tests/testPTHVD (hunk 6 above)
Comment (ShiboXing) [reply test/dlc_tests/container_tests/bin/pytorch_tests/testPTHVD:9]: Horovod has been found to break the latest Rubik binary, will send more info.

File: test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-adapt-BERT_pytorch-training-with-gloo-mpi-backend.patch (hunk 4 above)
Comment (ShiboXing) [reply test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-adapt-BERT_pytorch-training-with-gloo-mpi-backend.patch:2]: email info reverted

File: pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu (hunk 1 above)
Comment (saimidu) [reply pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu:490]: @ShiboXing Please define the Triton version required as an ARG next to the ARG that defines the SMMP binary if these are both interrelated packages.

Also, please add a comment mentioning that the Triton version installed in the image is dependent on the SMMP version.

File: pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu [hunk 7]
```diff
@@ -12,10 +12,10 @@ ARG PT_TORCHDATA_URL=https://download.pytorch.org/whl/test/torchdata-0.5.1-cp39-
 
 # SMD binaries
 ARG SMD_DATA_PARALLEL_URL=https://smdataparallel.s3.amazonaws.com/binary/pytorch/1.13.1/cu117/2023-01-09/smdistributed_dataparallel-1.7.0-cp39-cp39-linux_x86_64.whl
-ARG SMD_MODEL_PARALLEL_URL=https://sagemaker-distributed-model-parallel.s3.us-west-2.amazonaws.com/pytorch-1.13.1/build-artifacts/2023-01-19-18-35/smdistributed_modelparallel-1.14.0-cp39-cp39-linux_x86_64.whl
+ARG SMD_MODEL_PARALLEL_URL=https://sagemaker-distributed-model-parallel.s3.us-west-2.amazonaws.com/pytorch-1.13.1/build-artifacts/2023-04-17-15-49/smdistributed_modelparallel-1.15.0-cp39-cp39-linux_x86_64.whl
```
Comment (saimidu) [inline pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu:15]: Please define Triton version as an ARG here next to the SMMP URL ARG, and add a comment in the dockerfile mentioning that the Triton version is tied to the SMMP version, so that future reviewers can ensure that the correct version is installed.

File: test/dlc_tests/conftest.py [hunk 8]
```diff
@@ -618,6 +606,42 @@ def delete_s3_artifact_copy():
...
+def skip_s3plugin_test(request):
+    if "training" in request.fixturenames:
+        img_uri = request.getfixturevalue("training")
+    elif "pytorch_training" in request.fixturenames:
+        img_uri = request.getfixturevalue("pytorch_training")
+    else:
+        return
+    _, fw_ver = get_framework_and_version_from_tag(img_uri)
+    if request.node.get_closest_marker("skip_s3plugin_test"):
+        if Version(fw_ver) not in SpecifierSet("<=1.12.1,>=1.6.0"):
```
Comment (saimidu) [inline test/dlc_tests/conftest.py:619]: What happens if this `img_uri` was obtained from a `training` fixture for an MXNet 1.9 DLC?

If the `skip_s3plugin_test` fixture is meant to be autoused, why do you need to check if it is a marker on the pytest test function? If you need to check that the marker exists, then why not just make this fixture scoped to `session` instead of making it `autouse=True`, so that this function is only run when the marker is applied on a particular test?

File: test/dlc_tests/conftest.py [hunk 9]
```diff
@@ -618,6 +606,42 @@ def delete_s3_artifact_copy():
...
+            pytest.skip(
+                "s3 plugin is only supported in PT 1.6.0 - 1.12.1, skipping this container with tag{}".format(
+                    fw_ver
+                )
```
Comment (saimidu) [inline test/dlc_tests/conftest.py:623]: Why not just use f-strings, like:
```suggestion
                f"s3 plugin is only supported in PT 1.6.0 - 1.12.1, skipping this container with tag{fw_ver}"
```

It would make the code more compact and make the formatting easier to read.

File: test/dlc_tests/conftest.py [hunk 10]
```diff
@@ -618,6 +606,42 @@ def delete_s3_artifact_copy():
...
+@pytest.fixture(autouse=True)
+def skip_inductor_test(request):    
+    if "training" in request.fixturenames:
+        img_uri = request.getfixturevalue("training")
+    elif "pytorch_training" in request.fixturenames:
+        img_uri = request.getfixturevalue("pytorch_training")
+    else:
+        return
+    _, fw_ver = get_framework_and_version_from_tag(img_uri)
+    if request.node.get_closest_marker("skip_inductor_test"):
```
Comment (saimidu) [inline test/dlc_tests/conftest.py:636]: Same question as the `skip_s3plugin_test` fixture.

If the `skip_inductor_test` fixture is meant to be autoused, why do you need to check if it is a marker on the pytest test function? If you need to check that the marker exists, then why not just make this fixture scoped to `session` instead of making it `autouse=True`, so that this function is only run when the marker is applied on a particular test?

File: test/dlc_tests/conftest.py [hunk 11]
```diff
@@ -952,7 +976,7 @@ def generate_unique_values_for_fixtures(
         "autogluon": "ag",
     }
     fixtures_parametrized = {}
-
+    print(f"imgs: {images_to_parametrize}, {metafunc_obj.fixturenames}")
```
Comment (saimidu) [inline test/dlc_tests/conftest.py:979]: Is this meant to be a permanent addition to the code, or is it only here temporarily for the sake of debugging?

I don't see any output logs in the test jobs that contain `imgs: `

File: test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-adapt-BERT_pytorch-training-with-gloo-mpi-backend.patch (hunk 4 above)
Comment (saimidu) [reply test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-adapt-BERT_pytorch-training-with-gloo-mpi-backend.patch:2]: Does the commit where the email info change is reverted still need to be pushed?

File: test/dlc_tests/container_tests/bin/pytorch_tests/installOSSPyTorch [hunk 12]
```diff
@@ -3,9 +3,13 @@
...
-PT_VERSION=$(python -c "import torch; print(torch.__version__);")
-CUDA_VERSION=$(python -c "import torch; print(torch.version.cuda)")
+PT_VERSION=$(python -c "import torch; print(torch.__version__.split('+')[0]);")
+HAS_CUDA=$(python -c "import torch; print(torch.version.cuda is not None)")
 
 # install OSS PyTorch
-conda remove -y --force pytorch torchvision torchaudio \
-    && mamba install -y pytorch=$PT_VERSION torchvision torchaudio -c pytorch
+conda remove -y --force pytorch torchvision torchaudio 
+if [ -n "$cuda_version" ]; then
```
Comment (saimidu) [inline test/dlc_tests/container_tests/bin/pytorch_tests/installOSSPyTorch:11]: Should this be `$HAS_CUDA`? I don't see `cuda_version` being defined anywhere.

File: test/dlc_tests/container_tests/bin/pytorch_tests/installOSSPyTorch [hunk 13]
```diff
@@ -3,9 +3,13 @@
...
+    mamba install -y pytorch=$PT_VERSION torchvision torchaudio pytorch-cuda -c pytorch -c nvidia
+else
+    mamba install -y pytorch=$PT_VERSION torchvision torchaudio -c pytorch -c nvidia
```
Comment (saimidu) [inline test/dlc_tests/container_tests/bin/pytorch_tests/installOSSPyTorch:14]: If it's a non-GPU image, then why does `-c nvidia` need to be included?

File: test/dlc_tests/container_tests/bin/pytorch_tests/setupPyTorchBackendTest [hunk 14]
```diff
@@ -20,15 +29,6 @@ source ~/.bashrc
 # setup benchmark repo
 git clone https://github.com/pytorch/benchmark
 pushd benchmark
-git reset --hard 020aac26
+git reset --hard 78e54615
```
Comment (saimidu) [inline test/dlc_tests/container_tests/bin/pytorch_tests/setupPyTorchBackendTest:32]: How is this commit ID being decided? How should it be modified in the future?

Why does this commit ID need to be hardcoded in the test, instead of being dynamically defined based on something like PyTorch version in the image being tested, or something like that?

File: test/dlc_tests/container_tests/bin/pytorch_tests/testPTHVD (hunk 6 above)
Comment (saimidu) [reply test/dlc_tests/container_tests/bin/pytorch_tests/testPTHVD:9]: If Horovod breaks the latest SMMP binary, isn't that going to cause issues in the SMMP tests, rather than in the Horovod integ tests?

File: test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-adapt-BERT_pytorch-training-with-gloo-mpi-backend.patch (hunk 4 above)
Comment (ShiboXing) [reply test/dlc_tests/container_tests/bin/pytorch_tests/0001-test-adapt-BERT_pytorch-training-with-gloo-mpi-backend.patch]: just pushed

File: test/dlc_tests/container_tests/bin/pytorch_tests/testPTHVD (hunk 6 above)
Comment (ShiboXing) [reply test/dlc_tests/container_tests/bin/pytorch_tests/testPTHVD:9]: the problem is a bit complicated. So the installation of horovod and smmodelparallel binary + runtime of horovod (e.g. import), would trigger the illegal memory access of some sort. Previously the SMMP test was broken because of a horovod trigger in the smdebug. Thus with the upgraded smdebug, SMMP test is green now. Horovod test now needs fixing since the installation of SMMP is still present while horovod is imported.

File: test/dlc_tests/container_tests/bin/pytorch_tests/setupPyTorchBackendTest (hunk 14 above)
Comment (ShiboXing) [reply test/dlc_tests/container_tests/bin/pytorch_tests/setupPyTorchBackendTest:32]: Yes, I hope to pin this more properly in the future. Since the torch.compile is still new, we've been observing frequent errors if it's not pinned. And unfortunately, https://github.com/pytorch/benchmark doesn't have version tags for now. I've been manually updating the commit to latest every time I work on the backend test.

File: pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu [hunk 15]
```diff
@@ -486,7 +486,8 @@ RUN pip install --no-cache-dir -U \
     "sagemaker>=2,<3" \
     sagemaker-experiments==0.* \
     sagemaker-pytorch-training \
-    flash-attn==0.2.6.post1
+    flash-attn==0.2.8 \
```
[the remark this answers is not in the document]
Comment (ShiboXing) [reply pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu]: Yes it is required by the SMMP update, I've advised them to include these dependencies in their setup.py for the next version

File: pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu [hunk 16]
```diff
@@ -206,7 +206,7 @@ RUN pip install --no-cache-dir -U \
     boto3 \
     "click==8.1.2,<9" \
     "cmake>=3.24.3,<3.25" \
-    cryptography \
+    "cryptography>39,<40" \
```
[the remark this answers is not in the document]
Comment (ShiboXing) [reply pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu:212]: Unfortunately pyopenssl is 23.0.0 in this DLC, which depends on cryptography<40

File: pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu (hunk 15 above)
[the remark this answers is not in the document]
Comment (ShiboXing) [reply pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu]: parametrized and added comment

File: pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu [hunk 17]
```diff
@@ -541,6 +542,8 @@ RUN pip install --no-cache-dir -U ${SMD_MODEL_PARALLEL_URL}
 # Install SM Distributed DataParallel binary
 RUN SMDATAPARALLEL_PT=1 pip install --no-cache-dir ${SMD_DATA_PARALLEL_URL}
 
+# remove the default mamba installation
+RUN rm -rf /root/micromamba/
```
[the remark this answers is not in the document]
Comment (ShiboXing) [reply pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu:549]: unfortunately, CPU DLC is not built. I could add the changes in another PR later

File: pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu [hunk 18]
```diff
@@ -365,7 +365,7 @@ ARG RMM_VERSION=0.15.0
 
 # The smdebug pipeline relies for following format to perform string replace and trigger DLC pipeline for validating
 # the nightly builds. Therefore, while updating the smdebug version, please ensure that the format is not disturbed.
-ARG SMDEBUG_VERSION=1.0.25
+ARG SMDEBUG_VERSION=1.0.34
```
[the remark this answers is not in the document]
Comment (ShiboXing) [reply pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu:371]: unfortunately, CPU DLC is not built. I could add the changes in another PR later

File: test/dlc_tests/container_tests/bin/pytorch_tests/installOSSPyTorch (hunk 13 above)
Comment (ShiboXing) [reply test/dlc_tests/container_tests/bin/pytorch_tests/installOSSPyTorch]: fixed.

File: test/dlc_tests/conftest.py (hunk 8 above)
Comment (ShiboXing) [reply test/dlc_tests/conftest.py:619]: I noticed that a lot of the other skip_test fixtures have used autouse so I followed this convention. And since we checked the mark in the skip_test, the tests which have not been marked with this skip-test fixture will not be broken

File: test/dlc_tests/conftest.py (hunk 10 above)
Comment (ShiboXing) [reply test/dlc_tests/conftest.py:634]: I noticed that a lot of the other skip_test fixtures have used autouse so I followed this convention. And since we checked the mark in the skip_test, the tests which have not been marked with this skip-test fixture will not be broken

**Comment by arjkesh** [comment]
@ShiboXing do you mind updating to black==23.3.0 and running `black -l 100 .` on top level dir of your repo? This should take care of formatting issues

File: pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu [hunk 19]
```diff
@@ -9,13 +9,16 @@ ARG PT_SM_TRAINING_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.am
 ARG PT_TORCHVISION_URL=https://download.pytorch.org/whl/cu117/torchvision-0.14.1%2Bcu117-cp39-cp39-linux_x86_64.whl
 ARG PT_TORCHAUDIO_URL=https://download.pytorch.org/whl/cu117/torchaudio-0.13.1%2Bcu117-cp39-cp39-linux_x86_64.whl
 ARG PT_TORCHDATA_URL=https://download.pytorch.org/whl/test/torchdata-0.5.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
+# depended by the SMP binary
+ARG TRITON_VERSION=2.0.0.dev20221202
```
Comment (jeet4320) [inline pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu:13]: check if we need TRITON_VERSION

**Comment by viclzhu** [comment]
Upgrading SMMP to 1.15.0 for PT1.13 adds `Sharded Data Parallelism + Tensor Parallelism` support along with the associated full checkpointing support for SDP and `SDP + TP` through the existing checkpoint API. This requires changes in ZeRO-2D and SMMP for the support.

File: pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu (hunk 19 above)
Comment (ShiboXing) [reply pytorch/training/docker/1.13/py3/cu117/Dockerfile.gpu]: reverted!

**Review by jeet4320 — APPROVED** [review]
